js/private/repo_toolchains.bzl

```diff
@@ -0,0 +1,37 @@
+"Helpers to locate toolchains from within repository rules"
+
+load("@aspect_bazel_lib//lib:repo_utils.bzl", "repo_utils")
+
+def yq_path(rctx):
+    """Find the path to the host resolved yq binary from a repository rule.
+
+    Args:
+        rctx: The repository rule context
+
+    Returns:
+        Path to yq
+    """
+    return rctx.path(Label("{yq}{maybe_ext}".format(
+        yq = rctx.attr.yq,
+        maybe_ext = ".exe" if repo_utils.is_windows(rctx) and not rctx.attr.yq.endswith(".exe") else "",
+    )))
+
+def node_path(rctx):
+    """Find the path to the host resolved node binary from a repository rule.
+
+    Args:
+        rctx: The repository rule context
+
+    Returns:
+        Path to node
+    """
+
+    # Parse the resolved host platform from node host repo //:index.bzl
+    content = rctx.read(rctx.path(Label("@%s_host//:index.bzl" % rctx.attr.node_repository)))
+    search_str = "host_platform=\""
+    start_index = content.index(search_str) + len(search_str)
+    end_index = content.index("\"", start_index)
+    host_platform = content[start_index:end_index]
+
+    # Return the path to the node binary
+    return rctx.path(Label("@%s_%s//:bin/node%s" % (rctx.attr.node_repository, host_platform, ".exe" if repo_utils.is_windows(rctx) else "")))
```

Comment:
Once rnj is updated, it would be nice if we could just make a dumb function that takes the repo name and doesn't know about node or yq.

Reply:
Actually, maybe I'll just do that now and I'll leave the node path out of this until rnj is ready.

Reply:
yeah but I think they're necessarily different, because we need to add the _host on it

`rctx.path(Label("@{node}_host//:bin/node{maybe_ext}".format(`

Reply:
I guess they don't have to be though. You could change translate_pnpm_lock.bzl to use

```
"node": attr.string(
        doc = """The label to the node binary to use.""",
        default = "@nodejs_host//:bin/node",
    ),
    "yq": attr.string(
        doc = """The label to the yq binary to use. If executing on a windows host, the .exe extension will be appended if missing.""",
        default = "@yq//:yq",
    ),
```

and then the logic is the same

Reply:
Mm, yeah.